### PR TITLE
macOS VPN: Allow extension relocated in macOS 15 Sequoia

### PR DESCRIPTION
**Pull request by diegoreymendez**
Task/Issue URL: https://app.asana.com/0/1199230911884351/1208377130239025/f

## Description

Fixing the onboarding for macOS 15 Sequoia

## Known limitations

- The asset isn't final, it's just meant to be a quick fix for Sequoia users.  It's both single-size and light-mode only.

### Testing in macOS Sequoia

1. Uninstall the VPN either through Browser Settings > VPN, through Debug Menu > VPN > Reset > Reset all state, or by going into System Settings > General > Login Items and Extensions > Network Extensions > Uninstall our extension.
2. Start the VPN.
3. Make sure the first step in onboarding looks like this.
<div align="center">Screenshot 2024-09-24 at 3 28 25 PM</div>

4. Allow the extension and start the VPN.

### Testing in other macOS versions

1. Uninstall the VPN either through Browser Settings > VPN, through Debug Menu > VPN > Reset > Reset all state.
2. Start the VPN.
3. Make sure the first step in onboarding looks like this.
<div align="center">Screenshot 2024-09-24 at 3 36 51 PM</div>

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

File: LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/Extensions/UserText+NetworkProtectionUI.swift [hunk 1]
```diff
@@ -42,6 +42,9 @@ final class UserText {
     // MARK: - Onboarding
 
     static let networkProtectionOnboardingInstallExtensionTitle = NSLocalizedString("network.protection.onboarding.install.extension.title", value: "Install VPN System Extension", comment: "Title for the onboarding install-vpn-extension step")
+    static let networkProtectionOnboardingAllowExtensionDescPrefixForSequoia = NSLocalizedString("network.protection.onboarding.allow.extension.desc.sequoia.prefix", value: "Click ", comment: "Non-bold description for the onboarding allow-extension description")
+    static let networkProtectionOnboardingAllowExtensionDescEmphasized = NSLocalizedString("network.protection.onboarding.allow.extension.desc.sequoia.emphasized", value: "Open System Settings", comment: "'Allow' word between the prefix and suffix for the onboarding allow-extension description")
+    static let networkProtectionOnboardingAllowExtensionDescSuffixForSequoia = NSLocalizedString("network.protection.onboarding.allow.extension.desc.sequoia.suffix", value: ", then enable the DuckDuckGo VPN network extension.", comment: "Non-bold description for the onboarding allow-extension description")
```
Comment (diegoreymendez) [inline LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/Extensions/UserText+NetworkProtectionUI.swift:47]: Copy for sequoia only.

File: LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/NetworkProtectionAsset.swift [hunk 2]
```diff
@@ -49,6 +49,7 @@ public enum NetworkProtectionAsset: String, CaseIterable {
     case statusbarBrandedVPNIssueIcon = "statusbar-branded-vpn-issue"
 
     // Images
+    case enableSysexImage = "enable-sysex-bottom"
```
Comment (diegoreymendez) [inline LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/NetworkProtectionAsset.swift:52]: New image for the onboarding step.

File: LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/Views/PromptActionView/PromptActionView.swift [hunk 3]
```diff
@@ -78,7 +78,15 @@ struct PromptActionView: View {
             .padding(.horizontal, 10)
 
             if let actionScreenshot = model.actionScreenshot {
-                Image(actionScreenshot)
+                // This is done this way because the change was introduced as a hotfix
+                // for macOS Sequoia and we want to avoid breakage
+                if #available(macOS 15, *) {
```
Comment (diegoreymendez) [inline LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/Views/PromptActionView/PromptActionView.swift:83]: To minimize risk we onlychange things for macOS 15.  We can revisit this later on but not during a hotfix.

File: LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/Views/PromptActionView/PromptActionViewModel.swift [hunk 4]
```diff
@@ -114,11 +114,19 @@ extension OnboardingStep {
     func description(isMenuBar: Bool) -> [StyledTextFragment] {
         switch self {
         case .userNeedsToAllowExtension:
-            return [
-                .init(text: UserText.networkProtectionOnboardingAllowExtensionDescPrefix),
-                .init(text: UserText.networkProtectionOnboardingAllowExtensionDescAllow, isEmphasized: true),
-                .init(text: UserText.networkProtectionOnboardingAllowExtensionDescSuffix),
-            ]
+            if #available(macOS 15, *) {
```
Comment (diegoreymendez) [inline LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/Views/PromptActionView/PromptActionViewModel.swift:117]: Same principle, macOS 15 only changes.

File: LocalPackages/SystemExtensionManager/Sources/SystemExtensionManager/SystemExtensionManager.swift [hunk 5]
```diff
@@ -47,21 +47,9 @@ public struct SystemExtensionManager {
...
-        ///
-        ///     Tradeoffs: Unfortunately we don't know if the previous request was sent out by the currently runing-instance of this App
-        ///         or if an activation request was made, and then the App was reopened.
-        ///         This means we don't know if we'll be notified when the previous activation request completes or fails.  Because we
-        ///         need to update our UI once the extension is allowed, we can't avoid sending a new activation request every time.
-        ///         For the users that don't see the alert come up more than once this should be invisible.  For users (like myself) that
-        ///         see the alert every single time, they'll see both the alert and system settings being opened automatically.
-        ///
-        if hasPendingActivationRequests() {
-            openSystemSettingsSecurity()
```
Comment (diegoreymendez) [inline LocalPackages/SystemExtensionManager/Sources/SystemExtensionManager/SystemExtensionManager.swift:64]: The diff seems a bit complicated but it's really not.  I moved these comments and code into a dedicated method that's only run before macOS 15.  We no longer want to be trying to do what we're doing here in Sequoia.

For context this workaround is to try and open the right settings location for the user ourselves.  This is not great and can actually be really confusing when broken (like now).  Starting on Sequoia this is no longer done.

**Review by Bunn — COMMENTED** [review]
I'm seeing this on Sequioa 
image

Not sure if because of this:
```
(lldb) po onboardingStep
NetworkProtectionUI.OnboardingStep.userNeedsToAllowVPNConfiguration
(lldb) po onboardingStep.actionScreenshot
nil
```

But it doesn't look right

**Review by Bunn — APPROVED** [review]
LGTM, now what I saw was:
* Reset everyting
* First time clicking the button, I saw the correct macOS panel, clicking the "Open System Settings" took me to the correct place
* Then the following times clicking the button didn't take me to the same place as before, but to the "closest" place for the settings

**Pull request merged**